src/server/scripts/BrokenIsles/EyeOfAzshara/boss_wrath_of_azshara.cpp

```diff
@@ -15,10 +15,427 @@
+        }
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        BossAI::JustDied(killer);
+
+        DoCastSelf(SPELL_TIDESTONE_OF_GOLGANNETH, true);
+
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_FROST_RESONANCE);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_MAGIC_RESONANCE);
+
+        me->GetInstanceScript()->SetData(DATA_CRY_OF_WRATH, 2); // Remove Violent Winds
+    }
+
+    void ScheduleTasks() override
+    {
+        events.ScheduleEvent(SPELL_MASSIVE_DELUGE, 22s);
+        events.ScheduleEvent(SPELL_RAGING_STORMS, 5s);
+        events.ScheduleEvent(SPELL_TIDAL_WAVE_AT, 10s);
+        events.ScheduleEvent(SPELL_MYSTIC_TORNADO, 20s);
+        events.ScheduleEvent(SPELL_ARCANE_BOMB, 30s, 40s);
+
+        Difficulty instanceDifficulty = instance->instance->GetDifficultyID();
+        if (instanceDifficulty == DIFFICULTY_HEROIC || instanceDifficulty == DIFFICULTY_MYTHIC)
+            events.ScheduleEvent(SPELL_CRUSHING_DEPTHS, 25s, 45s);
+    }
+
+    void ExecuteEvent(uint32 eventId) override
+    {
+        switch (eventId)
+        {
+            case SPELL_MASSIVE_DELUGE:
+            {
+                if (Unit* target = SelectTarget(SELECT_TARGET_TOPAGGRO))
+                    me->CastSpell(target, SPELL_MASSIVE_DELUGE, false);
+                events.Repeat(25s);
+                break;
+            }
+            case SPELL_RAGING_STORMS:
+            {
+                bool hasPlayerInRange = false;
+                Map::PlayerList const& playerList = me->GetMap()->GetPlayers();
+                if (!playerList.isEmpty())
+                {
+                    for (Map::PlayerList::const_iterator itr = playerList.begin(); itr != playerList.end(); ++itr)
+                    {
+                        if (me->IsWithinMeleeRange(itr->GetSource()))
+                        {
+                            hasPlayerInRange = true;
+                            break;
+                        }
+                    }
+
+                    if (!hasPlayerInRange)
+                        DoCastSelf(SPELL_RAGING_STORMS, true);
+                }
+
+                events.Repeat(2s);
+                break;
+            }
+            case SPELL_MYSTIC_TORNADO:
+            {
+                DoCastSelf(SPELL_MYSTIC_TORNADO, false);
+                events.Repeat(20s, 30s);
+                break;
+            }
+            case SPELL_TIDAL_WAVE_AT:
+            {
+                DoCastSelf(SPELL_TIDAL_WAVE_AT, true);
+                if (_cryOfWrath)
+                    events.Repeat(5s);
+                else
+                    events.Repeat(10s, 15s);
+                break;
+            }
+            case SPELL_ARCANE_BOMB:
+                DoCastSelf(SPELL_ARCANE_BOMB, true);
+                events.Repeat(55s, 60s);
+                break;
+            case SPELL_CRUSHING_DEPTHS:
+                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1))
```

Comment:
This way it will select pets/guardians as valid targets as well iirc, unless that function changed and _onlyPlayer is true by default, which wasnt the case in the past - so worth taking a look